src/var/getShadowRoot.js

```diff
@@ -0,0 +1,15 @@
+define( function() {
+	"use strict";
+
+	return function getShadowRoot( obj ) {
+
+		// check for support of getRootNode function
+		if ( typeof obj.getRootNode === "function" ) {
+			return obj.getRootNode( { composed: true } );
+		} else {
+			var doc = obj.ownerDocument;
+			return doc && doc.host ? getShadowRoot( doc.host ) : doc;
```

Comment:
This will return the outer document for detached elements, which is incorrect. And we should probably have a loop instead of recursive calls anyway, but be mindful to return empty at all levels (e.g., including a shadow DOM inside a detached element).